src/CartesianProduct.jl

```diff
@@ -242,52 +216,15 @@ from the left.
 
 The modified Cartesian product of a finite number of convex sets.
 """
-function *(S::LazySet, cpa::CartesianProductArray)::CartesianProductArray
-    push!(cpa.array, S)
-    return cpa
+function CartesianProductArray(S::LazySet,
```

Comment:
ok, this change is to save the symbol `*` for the binary operator only, in other words these two that used to be the same are now different:

```julia
julia>  b = Ball2(zeros(2), 1.);

julia> c = CartesianProductArray([b, b])

julia> CartesianProductArray(b, c)
LazySets.CartesianProductArray{Float64,LazySets.Ball2{Float64}}(LazySets.Ball2{Float64}[LazySets.Ball2{Float64}([0.0, 0.0], 1.0), LazySets.Ball2{Float64}([0.0, 0.0], 1.0), LazySets.Ball2{Float64}([0.0, 0.0],
 1.0)])

julia> b * c
LazySets.CartesianProduct{Float64,LazySets.Ball2{Float64},LazySets.CartesianProductArray{Float64,LazySets.Ball2{Float64}}}(LazySets.Ball2{Float64}([0.0, 0.0], 1.0), LazySets.CartesianProductArray{Float64,Laz
ySets.Ball2{Float64}}(LazySets.Ball2{Float64}[LazySets.Ball2{Float64}([0.0, 0.0], 1.0), LazySets.Ball2{Float64}([0.0, 0.0], 1.0), LazySets.Ball2{Float64}([0.0, 0.0], 1.0)]))
```

i doubt a bit, but maybe it is more sensible.

Reply:
Actually I wanted to discuss this.
Previously `*` and `CartesianProduct` were not identical. I found this inconsistent, because `*(S, T)` was said to be an alias for `CartesianProduct(S, T)`.
Now it is identical, but we get what you observed.

I propose to additionally add a similar macro relating `CartesianProduct` and `CartesianProductArray` (and similarly for the other two) such that we also get the old behavior.
If you agree, should I do it here or in a follow-up PR?

Reply:
>>  follow-up PR

Reply:
Okay, I took a note in the related issue #162.